core/sql/common/charinfo.cpp

```diff
@@ -363,21 +363,22 @@ void CollationInfo::display() const
 CollationDB::CollationDB(CollHeap *h)
   : CollationDBSupertype(h), heap_(h), refreshNeeded_(TRUE)
 {
-    if (this == &CharInfo::builtinCollationDB_) return;
+    if (this == CharInfo::builtinCollationDB_) return;
```

Comment:
I'm confused. Is CollationDB a singleton object? Or is there one global version, and other parts of the code might create their own version?

Reply:
CharInfo::builtinCollationDB_ is a singleton object. It looks like there could other CollationDB objects, but it will have all the data in the builtinCollationDB object too